app/src/main/java/ch/epfl/favo/view/tabs/MapsPage.java

```diff
@@ -292,23 +279,20 @@ public View getInfoContents(Marker marker) {
 
   @Override
   public void onInfoWindowClick(Marker marker) {
-    // replaceFragment(new FavorDetailView(marker.getTitle(), marker.getSnippet()));
     if (marker.getTitle().equals(getString(R.string.self_location)))
       Navigation.findNavController(getView()).navigate(R.id.action_nav_map_to_favorRequestView);
-    // CommonTools.replaceFragment(
-    //    R.id.nav_host_fragment, getParentFragmentManager(), new FavorRequestView());
     else {
-      Favor favor = queryFavor(marker.getPosition().latitude, marker.getPosition().longitude);
+      Favor favor = queryFavor(marker.getPosition().latitude, marker.getPosition().longitude, marker.getTitle());
       Bundle favorBundle = new Bundle();
       favorBundle.putParcelable("FAVOR_ARGS", favor);
       Navigation.findNavController(getView())
           .navigate(R.id.action_nav_map_to_favorDetailView, favorBundle);
     }
   }
 
-  public Favor queryFavor(double latitude, double longitude) {
-    for (Favor favor : currentActiveLocalFavorList) {
-      if (favor.getLocation().getLatitude() == latitude
+  private Favor queryFavor(double latitude, double longitude, String title) {
```

Comment:
We can maybe change the name to something clearer like findFavorInLocalList().

Reply:
Actually, I was wondering if you could append the favor Id in the marker data, so that you can query the list with the Id directly, rather than using the location

Reply:
done